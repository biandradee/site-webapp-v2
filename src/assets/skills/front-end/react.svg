<svg width="80" height="80" viewBox="0 0 80 80" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M60.3601 39.9997C60.3601 44.8201 51.243 48.727 39.9964 48.727C28.7499 48.727 19.6328 44.8201 19.6328 39.9997C19.6328 35.1794 28.7499 31.2725 39.9964 31.2725C51.243 31.2725 60.3601 35.1794 60.3601 39.9997Z" stroke="white" stroke-width="2"/>
<path d="M30.302 57.9052C34.5405 60.2034 42.3165 54.0448 47.6751 44.1568C53.0307 34.2659 53.9383 24.3896 49.6998 22.0943C45.4613 19.7961 37.6853 25.9546 32.3267 35.8426C26.9711 45.7336 26.0634 55.6099 30.302 57.9052Z" stroke="white" stroke-width="2"/>
<path d="M49.6998 57.9052C45.4613 60.2034 37.6853 54.0448 32.3267 44.1568C26.9711 34.2659 26.0634 24.3896 30.302 22.0943C34.5405 19.7961 42.3165 25.9546 47.6751 35.8426C53.0307 45.7336 53.9383 55.6099 49.6998 57.9052Z" stroke="white" stroke-width="2"/>
<path d="M37.0859 39.9999C37.0859 40.7715 37.3924 41.5114 37.938 42.057C38.4836 42.6025 39.2235 42.909 39.995 42.909C40.7666 42.909 41.5065 42.6025 42.0521 42.057C42.5976 41.5114 42.9041 40.7715 42.9041 39.9999C42.9041 39.2284 42.5976 38.4884 42.0521 37.9429C41.5065 37.3973 40.7666 37.0908 39.995 37.0908C39.2235 37.0908 38.4836 37.3973 37.938 37.9429C37.3924 38.4884 37.0859 39.2284 37.0859 39.9999Z" stroke="white" stroke-width="2"/>
</svg>
