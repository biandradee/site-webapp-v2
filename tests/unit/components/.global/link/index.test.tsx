import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Link from "../../../../../src/components/.global/link";
import "@testing-library/jest-dom";
import { expectWithPNG } from "../../../../setup/simplePngMatcher";

describe("Link Component", () => {
  // Timeout maior para testes visuais
  jest.setTimeout(15000);

  it("should render correctly and handle click", () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Link onClick={handleClick}>Test Link</Link>);

    const linkElement = getByText("Test Link");
    expect(linkElement).toBeInTheDocument();

    fireEvent.click(linkElement);
    expect(handleClick).toHaveBeenCalled();
  });

  it("should match the screenshot and generate PNG automatically", async () => {
    const { container } = render(
      <div
        style={{
          padding: "20px",
          backgroundColor: "#f8f9fa",
          display: "inline-block",
          minWidth: "200px",
          borderRadius: "8px",
        }}
      >
        <Link onClick={() => {}}>Test Link</Link>
      </div>
    );

    // Esta linha automaticamente gera um PNG E faz o snapshot
    await expectWithPNG(container, "link-default-state");
  });

  it("should generate PNG for Link hover state", async () => {
    const { container } = render(
      <div
        style={{
          padding: "20px",
          backgroundColor: "#f8f9fa",
          display: "inline-block",
          minWidth: "200px",
          borderRadius: "8px",
        }}
      >
        <Link onClick={() => {}}>Hover Link</Link>
      </div>
    );

    // Simula o estado hover
    const linkElement = container.querySelector("a");
    if (linkElement) {
      linkElement.style.textDecoration = "underline";
    }

    // Gera PNG automaticamente
    await expectWithPNG(container, "link-hover-state");
  });

  it("should generate PNG for Link with different text", async () => {
    const { container } = render(
      <div
        style={{
          padding: "20px",
          backgroundColor: "#f8f9fa",
          display: "inline-block",
          minWidth: "300px",
          borderRadius: "8px",
        }}
      >
        <Link onClick={() => {}}>Este é um link mais longo para teste</Link>
      </div>
    );

    // Gera PNG automaticamente
    await expectWithPNG(container, "link-long-text");
  });

  it("should generate PNG for Link with custom styling", async () => {
    const { container } = render(
      <div
        style={{
          padding: "30px",
          backgroundColor: "#e3f2fd",
          display: "inline-block",
          minWidth: "200px",
          borderRadius: "8px",
          boxShadow: "0 4px 6px rgba(0,0,0,0.1)",
        }}
      >
        <Link onClick={() => {}}>Link Customizado</Link>
      </div>
    );

    // Gera PNG automaticamente
    await expectWithPNG(container, "link-custom-styling");
  });

  it("should generate PNG for multiple Links", async () => {
    const { container } = render(
      <div
        style={{
          padding: "20px",
          backgroundColor: "#f8f9fa",
          display: "flex",
          gap: "15px",
          flexWrap: "wrap",
          minWidth: "400px",
          borderRadius: "8px",
        }}
      >
        <Link onClick={() => {}}>Home</Link>
        <Link onClick={() => {}}>Sobre</Link>
        <Link onClick={() => {}}>Contato</Link>
        <Link onClick={() => {}}>Serviços</Link>
      </div>
    );

    // Gera PNG automaticamente
    await expectWithPNG(container, "link-multiple-links");
  });
  // Teste que usa container.toMatchSnapshot() tradicional + PNG
  it("should use traditional toMatchSnapshot with PNG generation", async () => {
    const { container } = render(
      <div
        style={{
          padding: "25px",
          backgroundColor: "#fff3e0",
          display: "inline-block",
          minWidth: "250px",
          borderRadius: "12px",
          border: "2px solid #ff9800",
        }}
      >
        <Link onClick={() => {}}>Link com Border</Link>
      </div>
    );

    // Gera PNG primeiro
    await expect(container).toMatchPNGSnapshot();

    // Depois faz o snapshot tradicional
    expect(container.innerHTML).toMatchSnapshot();
  });
});
