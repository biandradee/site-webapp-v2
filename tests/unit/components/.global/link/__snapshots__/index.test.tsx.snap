// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Link Component should generate PNG for Link hover state 1`] = `"<div style="padding: 20px; background-color: rgb(248, 249, 250); display: inline-block; min-width: 200px; border-radius: 8px;"><a class="sc-bRKDuR hsgkbQ" style="text-decoration: underline;">Hover Link</a></div>"`;

exports[`Link Component should generate PNG for Link with custom styling 1`] = `"<div style="padding: 30px; background-color: rgb(227, 242, 253); display: inline-block; min-width: 200px; border-radius: 8px; box-shadow: 0 4px 6px rgba(0,0,0,0.1);"><a class="sc-bRKDuR hsgkbQ">Link Customizado</a></div>"`;

exports[`Link Component should generate PNG for Link with different text 1`] = `"<div style="padding: 20px; background-color: rgb(248, 249, 250); display: inline-block; min-width: 300px; border-radius: 8px;"><a class="sc-bRKDuR hsgkbQ">Este é um link mais longo para teste</a></div>"`;

exports[`Link Component should generate PNG for multiple Links 1`] = `"<div style="padding: 20px; background-color: rgb(248, 249, 250); display: flex; gap: 15px; flex-wrap: wrap; min-width: 400px; border-radius: 8px;"><a class="sc-bRKDuR hsgkbQ">Home</a><a class="sc-bRKDuR hsgkbQ">Sobre</a><a class="sc-bRKDuR hsgkbQ">Contato</a><a class="sc-bRKDuR hsgkbQ">Serviços</a></div>"`;

exports[`Link Component should match the screenshot and generate PNG automatically 1`] = `"<div style="padding: 20px; background-color: rgb(248, 249, 250); display: inline-block; min-width: 200px; border-radius: 8px;"><a class="sc-bRKDuR hsgkbQ">Test Link</a></div>"`;

exports[`Link Component should use traditional toMatchSnapshot with PNG generation 1`] = `"<div style="padding: 25px; background-color: rgb(255, 243, 224); display: inline-block; min-width: 250px; border-radius: 12px; border: 2px solid rgb(255, 152, 0);"><a class="sc-bRKDuR hsgkbQ">Link com Border</a></div>"`;
